[feat] 강아지 이름으로 조회 기능 추가, 강아지 정보 통째로 덮어쓰는 기능 추가

2ternal opened the pull request:
## 기능구현
- 강아지 이름으로 강아지 조회하는 기능 구현
![image](https://user-images.githubusercontent.com/26919446/113503435-cb045f00-956c-11eb-9799-db50f988d9b9.png)

  - 강아지 이름으로 존재하지 않는경우 404에러
 
![image](https://user-images.githubusercontent.com/26919446/113503454-ec654b00-956c-11eb-8ba5-01d5069256c3.png)

- 강아지 정보를 통째로 덮어쓰는 기능 구현
  - 진료 기록을 받지 않은 경우
![image](https://user-images.githubusercontent.com/26919446/113503483-1b7bbc80-956d-11eb-8ecb-73aa1ca617e9.png)
  - 진료 기록을 받았지만 정보가 일치하는 경우
![image](https://user-images.githubusercontent.com/26919446/113503547-8a591580-956d-11eb-8daa-2b72c98ed720.png)


  - 수정하려는 강아지의 이름이 존재하지 않을경우 404에러
![image](https://user-images.githubusercontent.com/26919446/113503499-45cd7a00-956d-11eb-94dd-5cd82a1e2374.png)

  - 진료기록 수정 혹은 기존의 진료기록과 다를경우 403에러
![image](https://user-images.githubusercontent.com/26919446/113503515-64337580-956d-11eb-963b-13e8be488282.png)


- git이 처음이라 한번에 너무 많은 양을 commit, push해서 죄송합니다.

2ternal committed:
[feat]강아지 이름으로 강아지를 조회하는 기능 추가, 강아지 정보 통째로 덮어쓰는 기능 추가
-getDogName Controller구현
-getDogName Service구현
-getDogName Repository구현

-modifyWithAll Controller구현
-modifyWithAll Service구현
-modifyWithAll Repository구현

2ternal committed:
[feat]DogForbiddenException 에러 추가
- 강아지 진료기록을 수정하려할 때 반환하는 403에러 추가

qkrdmstlr3 commented on src/main/java/org/cnu/realcoding/repository/DogRepository.java:
```diff
@@ -13,4 +14,24 @@
+
+    public Dog modifyWithAll(Dog dog, Dog modDog) {
+
+        if(modDog.getMedicalRecords() == null || Arrays.equals(modDog.getMedicalRecords().toArray(), dog.getMedicalRecords().toArray())){
```
길어서 식별하기 힘든 조건문은 Boolean isMedicalRecordsAuthorized = 뭐 이런식으로 변수로 빼서 넣어주는게 가독성에 더 좋을것 같아요!

2ternal replied:
그렇네요. 다음부터는 가독성도 신경써보도록 하겠습니다!

qkrdmstlr3 reviewed (approved):
수고하셨습니다!

bardisue commented on src/main/java/org/cnu/realcoding/exception/DogNotFoundException.java:
```diff
@@ -0,0 +1,10 @@
+package org.cnu.realcoding.exception;
```
에러처리를 하는 exception을 두개의 자바파일로 나누셨는데 하나로 통합하는 의견에 대해서는 어떻게 생각하시나요?

2ternal replied:
이미 존재하는 exception 자바 파일명이 DogNotFoundException(존재 하지 않을 때 발생)이라,  다른 자바 파일로 DogForbiddenException(요청에 대한 권한이 없을 때 발생)을 생성해 줬었는데 만약 파일명을 포괄적인 내용을 담는 이름으로 변경한다면 통합해서 써도 괜찮을거같네요!

bardisue replied:
다시 생각해봤는데 두개를 통합할 방법이 잘 떠오르질 않네요 이와 같은 방법으로 가도 좋을것 같습니다.

2ternal committed:
[refactor]Mongodb연결/그에 따른 강아지 생성, 강아지 이름으로 조회, 특정 강아지 정보 수정 등
- 강아지 생성 DogRepository.java의 insertDog는 '박병준'의 코드를 따옴
- 이름, 주인이름, 주인전화번호로 강아지를 찾는 DogRepository.java의 getDogByThreeParams는 '연제령'의 코드를 따옴
- getDogName의 이름을 getDogByDogName으로 수정
- getDogByDogName이 같은 이름의 강이지들를 리턴할 수 있도록 수정
- modifyWithAll을 mongdb에 연동
-- 수정시 유니크키와 겹치는지 확인하는 기능이 필요

bardisue committed:
Merge branch 'dev' into feature/권우철
